### PR TITLE
26808 - Feature - Add support to allow Private DNS Zone VNet Linking for Spoke VNets

### PR DESCRIPTION
**Pull request by JamJarchitect**
# Overview/Summary

Adding support to allow Private DNS Zone VNet Linking for Spoke VNets through HubPeeredSpoke orchestration module and a new Private Dns Zone Linking module.

[AB#26808](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/26808)

## This PR fixes/adds/changes/removes

1. Added feature to HubPeeredSpoke orchestration module to allow Private DNS Zone Linking to Spoke VNets.
2. Added new module for Private DNS Zone Linking.

### Breaking Changes

1. N/A

## Testing Evidence

Evidence to be posted.

## As part of this Pull Request I have

- [X] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [X] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [X] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [X] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [X] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by JamJarchitect** [comment]
image

New module working - orchestration module test to be done

File: infra-as-code/bicep/orchestration/hubPeeredSpoke/bicepconfig.json [hunk 1]
```diff
@@ -57,32 +57,8 @@
...
-          "level": "error"
-        },
-        "use-recent-api-versions":{
-          "level": "error"
-        },
-        "use-resource-id-functions":{
-          "level": "error"
-        },
-        "use-stable-resource-identifiers":{
-          "level": "error"
```
Comment (jtracey93) [inline infra-as-code/bicep/orchestration/hubPeeredSpoke/bicepconfig.json:83]: Can we put these back?

File: infra-as-code/bicep/modules/privateDnsZoneLinks/samples/generateddocs/baseline.sample.bicep.md
Comment (jtracey93) [inline infra-as-code/bicep/modules/privateDnsZoneLinks/samples/generateddocs/baseline.sample.bicep.md:1]: This file shouldnt exist right?

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Looking good, couple of small changes please @JamJarchitect

File: infra-as-code/bicep/modules/privateDnsZoneLinks/samples/generateddocs/baseline.sample.bicep.md
Comment (JamJarchitect) [reply infra-as-code/bicep/modules/privateDnsZoneLinks/samples/generateddocs/baseline.sample.bicep.md:1]: We have a samples folder for most modules with a generateddocs folder and markdowns.

Shall I create another task to look at these to exclude samples for generated markdowns after closing this one?

File: infra-as-code/bicep/orchestration/hubPeeredSpoke/bicepconfig.json (hunk 1 above)
Comment (JamJarchitect) [reply infra-as-code/bicep/orchestration/hubPeeredSpoke/bicepconfig.json]: Added back

File: infra-as-code/bicep/modules/privateDnsZoneLinks/samples/generateddocs/baseline.sample.bicep.md
Comment (jtracey93) [reply infra-as-code/bicep/modules/privateDnsZoneLinks/samples/generateddocs/baseline.sample.bicep.md:1]: my apologies, ignore this then

**Pull request closed without merging**